Documentation/proposals/OneDotNetMSBuildProperties.md

```diff
@@ -0,0 +1,178 @@
+when (and if) they are removed from project templates.
+
+As a starting point, we can add the following MSBuild properties to
+the Xamarin.Android and Xamarin.iOS SDKs:
+
+* `$(ApplicationId)` maps to `/manifest/@package` and
+  `CFBundleIdentifier`
+* `$(ApplicationVersion)` maps to `android:versionName` or
+  `CFBundleVersion`. This is a version string that must be incremented
+  for each iOS App Store or TestFlight submission.
+* `$(ApplicationVersionCode)` maps to `android:versionCode` (_Android
+  only)_. This is unfortunately an integer and must be incremented for
+  each Google Play submission. If omitted, the Android SDK can use
+  `$(ApplicationVersion)` for values up to 128.128.128.128. We can
+  pack most version strings into a 32-bit integer by default.
+* `$(ApplicationDisplayVersion)` maps to `CFBundleShortVersionString`
+  (_iOS only)_. This can default to `$(ApplicationVersion)` when
+  blank.
+* `$(ApplicationTitle)` maps to `/application/@android:title` or
+  `CFBundleDisplayName`
+
+The final value that is generated in the `Info.plist` or
+`AndroidManifest.xml` can be overridden at different times. The final
+source of truth is determined in order of:
+
+1. `Info.plist` or `AndroidManifest.xml` in the iOS/Android head project.
+2. iOS/Android head `.csproj` defines the MSBuild properties
+3. _(To be implemented in MAUI/Forms)_ set in a shared `.csproj`.
+4. The properties set by MSBuild via other means such as
+   `Directory.Build.props`, etc.
+
+Even if we did not complete the goal of complete removal of
+`AndroidManifest.xml` and `Info.plist` from Xamarin project templates,
+these new MSBuild properties would be useful in their own right.
+
+## Opting out
+
+.NET Core introduced `$(GenerateAssemblyInfo)` which defaults to `true`.
+Projects migrating to .NET Core might set this to `false` if they have
+an existing `Properties/AssemblyInfo.cs`. We should have a similar
+property to disable the behavior.
+
+`$(GenerateApplicationManifest)` defaults to `true` in .NET 6 and
+`false` in "legacy" Xamarin.Android/Xamarin.iOS.
+
+In most cases, developers would only use `$(GenerateApplicationManifest)`
+if they want to try the new features in "legacy" Xamarin.
+
+## Android Template
+
+The default Android project template would include:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net6.0-android</TargetFramework>
+    <ApplicationTitle>MyApp</ApplicationTitle>
+    <ApplicationId>com.companyname.myapp</ApplicationId>
+    <ApplicationVersion>1.0</ApplicationVersion>
+  </PropertyGroup>
+</Project>
+```
+
+`$(ApplicationVersionCode)` is computed by a math expression to
+translate versions up to 128.128.128.128 to a 32-bit integer.
+
+Removed from `AndroidManifest.xml` in the project template:
+
+* `/manifest/@android:versionCode="1"`
+* `/manifest/@android:versionName="1.0"`
+* `/manifest/@package="com.companyname.myapp"`
+* `/application/@android:label="MyApp"`
```

Comment:
Are we OK that the removal of these will make this property page blank? (until there is IDE support)

![image](https://user-images.githubusercontent.com/840039/101692304-a7556000-3a35-11eb-8bc2-935e91f60475.png)

Reply:
Do we know what the time delay would be between SDK support for these properties *and release* vs. IDE support?  If the IDE supports the new semantics before .NET 6 P1, it'll be fine.  If the IDE won't catch up until P2, things could be "weird".

Reply:
I don't think we know yet.

We could always implement the new MSBuild properties, but only add them to the project templates when the IDE-side lands.